monai/data/image_reader.py

```diff
@@ -626,3 +630,143 @@ def _get_spatial_shape(self, img) -> np.ndarray:
+
+    def read(self, data: Union[Sequence[str], str, np.ndarray], **kwargs):
+        """
+        Read image data from specified file or files.
+        Note that the returned object is CuImage or list of CuImage objects.
+
+        Args:
+            data: file name or a list of file names to read.
+
+        """
+        img_: List = []
+
+        filenames: Sequence[str] = ensure_tuple(data)
+        for name in filenames:
+            img = self.wsi_reader(name)
+            if self.wsi_reader_name == "openslide":
+                img.shape = (img.dimensions[1], img.dimensions[0], 3)
+            img_.append(img)
+
+        return img_ if len(filenames) > 1 else img_[0]
+
+    def get_data(
+        self,
+        img_obj,
+        location: Tuple[int, int] = (0, 0),
+        size: Optional[Tuple[int, int]] = None,
+        level: int = 0,
+        dtype: DtypeLike = np.uint8,
+        grid_shape: Tuple[int, int] = (1, 1),
+        patch_size: Optional[int] = None,
+    ):
+        """
+        Extract regions as numpy array from WSI image and return them.
+
+        Args:
+            img:      a wsi_reader object loaded from a file, or list of CuImage objects
+            location: (x_min, y_min) tuple giving the top left pixel in the level 0 reference frame,
+                       or list of tuples (default=(0, 0))
+            size:     (height, width) tuple giving the region size, or list of tuples (default=(wsi_width, wsi_height))
+                        This is the size of image at the given level (`level`)
+            level:    the level number, or list of level numbers (default=0)
+
+        """
+        if size is None:
+            if location == (0, 0):
+                # the maximum size is set to WxH
+                size = (img_obj.shape[0] // (2 ** level), img_obj.shape[1] // (2 ** level))
+                print(f"Reading the whole image at level={level} with shape={size}")
+            else:
+                print("Size need to be provided!")
+                return
+        region = self._extract_region(img_obj, location=location, size=size, level=level, dtype=dtype)
+        if patch_size is None:
+            patches = region
+        else:
+            patches = self._extract_patches(
+                region, patch_size=(patch_size, patch_size), grid_shape=grid_shape, dtype=dtype
+            )
+        return patches
+
+    def _extract_region(
+        self,
+        img_obj,
+        size: Tuple[int, int],
+        location: Tuple[int, int] = (0, 0),
+        level: int = 0,
+        dtype: DtypeLike = np.uint8,
+    ):
+        # reverse the order of dimensions for size and location to be compatible with image shape
+        size = size[::-1]
+        location = location[::-1]
+        region = img_obj.read_region(location=location, size=size, level=level)
+        if self.wsi_reader_name == "openslide":
+            region = region.convert("RGB")
+        # convert to numpy
```

Comment:
I would suggest to convert the data shape in the following `EnsureChannelFirst` transform later, don't hard-code here. @wyli @rijobro , what do you think?

Reply:
this channel last format is a very common default, perhaps we could embed an `EnsureChannelFirst` here (or for all the readers)? otherwise the user almost always need to add an `EnsureChannelFirst`

Reply:
Hi @wyli ,

Interesting idea, we can do it later as it's a non-breaking feature, `EnsureChannelFirst` transform will not cause an error if the data is already channel-first in readers.

Thanks.

Reply:
But for current PR, I think it should not change the data shape here.
If we want to use `EnsureChannelFirst`, we should apply in the base class.

Thanks.

Reply:
I think for any reader that we are implementing, we should make sure that it is channel first, to follow the convention throughout the readers.